libraw/src/main/java/com/homesoft/photo/libraw/LibRaw.java

```diff
@@ -260,18 +265,90 @@ public Map<Float, float[]> getAvailableWhiteBalanceCoefficientsWithTemps() {
     @RequiresApi(26)
     private native boolean drawHardwareBuffer(HardwareBuffer hardwareBuffer);
 
-    @RequiresApi(Build.VERSION_CODES.Q)
-    private Bitmap getHardwareBitmap(int format) {
+    /**
+     * Decode to a HardwareBuffer
+     * @param format must be {@link HardwareBuffer#RGB_888} or {@link HardwareBuffer#RGBA_FP16}
+     * @return null if no image is available
+     */
+    @RequiresApi(26)
+    @Nullable
+    public HardwareBuffer getHardwareBuffer(int format) {
```

Comment:
this is nice, I love the ability to just get a buffer rather than a bitmap. will make passing it around way easier.

Reply:
I thought you said you had a need for this, so I exposed it.

Reply:
yep, thank you!